pallets/mnt-token/src/lib.rs

```diff
@@ -209,17 +213,22 @@ pub mod module {
 				MntSpeeds::<T>::insert(currency_id, Balance::zero());
 				MntPoolsState::<T>::insert(currency_id, MntPoolState::new());
 			}
-			if !self.minted_pools.is_empty() {
-				Pallet::<T>::refresh_mnt_speeds().expect("Calculate MntSpeeds is failed");
-			}
 		}
 	}
 
 	#[pallet::pallet]
 	pub struct Pallet<T>(PhantomData<T>);
 
 	#[pallet::hooks]
-	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_finalize(block: T::BlockNumber) {
+			if block % T::SpeedRefreshPeriod::get() == T::BlockNumber::zero() {
+				if let Err(msg) = Pallet::<T>::refresh_mnt_speeds() {
+					debug::error!("{:?}", msg)
+				}
+			}
+		}
```

Comment:
I think it can be more informative.
```suggestion
		fn on_finalize(block: T::BlockNumber) {
			if (block % T::SpeedRefreshPeriod::get()).is_zero() {
				if let Err(msg) = Pallet::<T>::refresh_mnt_speeds() {
					debug::error!(
						"MntToken on_finalize(). Cannot run refresh_mnt_speed() at {:?}: {:?}",
						block,
						msg
					)
				}
			}
		}
```